dango/dex/src/execute.rs

```diff
@@ -209,14 +222,16 @@ pub fn cron_execute(ctx: SudoCtx) -> StdResult<Response> {
     let mut events = Vec::new();
     let mut refunds = BTreeMap::new();
 
-    // Find all pairs that have received new orders during the block.
-    let pairs = NEW_ORDER_COUNTS
-        .current_range(ctx.storage, None, None, IterationOrder::Ascending)
-        .map(|res| {
-            let (pair, _) = res?;
-            Ok(pair)
-        })
+    // Add all incoming orders to the orders map and clear the incoming orders map.
+    let mut pairs = BTreeSet::new();
+    let incoming_orders = INCOMING_ORDERS
+        .range(ctx.storage, None, None, IterationOrder::Ascending)
         .collect::<StdResult<Vec<_>>>()?;
+    for (_, (order_key, order)) in incoming_orders {
+        ORDERS.save(ctx.storage, order_key.clone(), &order)?;
+        pairs.insert(order_key.0);
+    }
+    INCOMING_ORDERS.clear(ctx.storage, None, None);
```

Comment:
I think this part of the code is quite ugly.

First, let's don't use `order_key.0`. Do something like:

```rust
for (_, ((pair, direction, price, order_id), order)) in incoming_orders {
    ORDERS.save(ctx.storage, (pair.clone(), direction, price, order_id), &order)?;
    pairs.insert(pair);
}
```

Sure it's more verbose, but a lot more readable imo.

Second, we have a `Map::take` method that does `Map::load` and `Map::remove` in one go. We should add a similar method `Map::drain` that does `Map::range` and `Map::clear` in one go.

Third, instead of pushing items into `pairs`, it's better to do something like:

```rust
let pairs = incoming_orders.keys().map(/* ... */).collect();
```

Reply:
Actually you don't even need to collect `pairs` into a btreeset. You can simply do:

```rust
for (pairs, ..) in incoming_orders.keys() {
    // order clearing logics for each pair...
}
```

Reply:
See: https://github.com/left-curve/left-curve/compare/main...drain